### PR TITLE
Additional CLI commands

### LINKED ISSUE #2045: Improve hyperspace CLI
Extra CLI commands are required in hyperspace for the following
- Setting up clients
- Connections
- Channels

These commands need to be available for deployment.

### PR DESCRIPTION
**Pull request by Wizdave97**
closes #2045

### CODE REVIEW

File: code/centauri/hyperspace/bin/main.rs [hunk 1]
```diff
@@ -19,19 +38,32 @@ pub struct Cli {
 /// Possible subcommands of the main binary.
 #[derive(Debug, Parser)]
 pub enum Subcommand {
-	Relay(RelayCmd),
+	#[clap(name = "relay", about = "Start relaying messages between two chains")]
+	Relay(Cmd),
```
Comment (blasrodri) [inline code/centauri/hyperspace/bin/main.rs:42]: now that we've added more commands, can we be more specific on what relay does?

**Review by blasrodri — APPROVED** [review]

**Review by vmarkushin — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

File: code/centauri/hyperspace/bin/main.rs [hunk 2]
```diff
@@ -5,10 +5,29 @@ use metrics::{data::Metrics, handler::MetricsHandler, init_prometheus};
...
+use ibc::core::ics04_channel::channel::{ChannelEnd, Order, State};
+use ibc::core::ics04_channel::msgs::chan_open_init::MsgChannelOpenInit;
+use ibc::core::ics24_host::identifier::PortId;
+use ibc::events::IbcEvent;
+use ibc::tx_msg::Msg;
+use ibc_proto::google::protobuf::Any;
+use primitives::{IbcProvider, KeyProvider};
+use primitives::mock::LocalClientTypes;
+use primitives::utils::timeout_future;
+use futures::StreamExt;
```
Comment (seunlanlege) [inline code/centauri/hyperspace/bin/main.rs:30]: run `cargo fmt`

File: code/centauri/hyperspace/parachain/src/lib.rs [hunk 3]
```diff
@@ -389,10 +408,203 @@ where
...
+		T::Header: HeaderT,
+		<T::Header as HeaderT>::Hash: From<T::Hash>,
+		T::BlockNumber: From<u32>,
+		FinalityProof<sp_runtime::generic::Header<u32, sp_runtime::traits::BlakeTwo256>>:
+		From<FinalityProof<T::Header>>,
+		BTreeMap<H256, ParachainHeaderProofs>:
+		From<BTreeMap<<T as Config>::Hash, ParachainHeaderProofs>>,
+{
+	/// Construct a beefy client state to be submitted to the counterparty chain
+	pub async fn construct_beefy_client_state(
```
Comment (seunlanlege) [inline code/centauri/hyperspace/parachain/src/lib.rs:441]: Feels like we should add this method to `Chain` as `initialize_client_state`

File: code/centauri/hyperspace/parachain/src/provider.rs [hunk 4]
```diff
@@ -588,4 +591,32 @@ where
 		let refresh_period: u64 = if cfg!(feature = "testing") { 15 } else { 50 };
 		latest_height - latest_client_height_on_counterparty >= refresh_period
 	}
+
+	async fn construct_client_state(&self) -> Result<(AnyClientState, AnyConsensusState), Self::Error> {
```
Comment (seunlanlege) [inline code/centauri/hyperspace/parachain/src/provider.rs:595]:
```suggestion
	async fn initialize_client_state(&self) -> Result<(AnyClientState, AnyConsensusState), Self::Error> {
```

**Comment by seunlanlege** [comment]
Suggest that we consolidate client initialization and connection/channel negotiation code with the testsuite and have them in primitives. @Wizdave97

**Comment by Wizdave97** [comment]
> Suggest that we consolidate client initialization and connection/channel negotiation code with the testsuite and have them in primitives. @Wizdave97

Working on it.

**Review by seunlanlege — APPROVED** [review]

**Pull request merged**